dbc2val/dbcfeederlib/dbcparser.py

```diff
@@ -34,16 +34,20 @@ def __init__(self, dbcfile: str, use_strict_parsing: bool = True):
         for name in dbcfile.split(","):
             filename = name.strip()
             if filename in found_names:
-                log.warn("The DBC file {} has already been read, ignoring it!".format(filename))
+                log.warning("The DBC file {} has already been read, ignoring it!".format(filename))
                 continue
             found_names.add(filename)
             if first:
```

Comment:
Cooudl this logic with "if first" etc. be simplified by creating a database and just adding to it. There seems to be a function `add_dbc(fp)` (or add_dbc_file) and the documentation has this example
```
>>> db = cantools.database.Database()
>>> with open ('foo.dbc', 'r') as fin:
...     db.add_dbc(fin)
```

That would not need a temperary db, that is serialized and then parsed again
I see this probably kills the "automagically determine database type", but there are also e.g. add_kcd functions, if we would try determine ourselves what is the correct one tho choose

Anyway, this is a bit of "armchair coding", I did not test, the code here just did not strike me as super elegant and efficient, and especially on slower targets we found DBC parsing takes a while....

Reply:
That is possible, of course. The *database* module has a (private) function for determining the database file type which we could copy. If you are that concerned about performance, I can change it accordingly. 

What is the actual use case for loading multiple database files anyway? Is the idea to have one (big) database file that contains the *standard* messages and you then apply some *diffs* via additional files?

Reply:
Anyway, I have pushed a corresponding change ...

Reply:
We have setups, where we need to interact with separate components/ECUs on a single CAN Bus, but where DBC description are delivered "per component". Of course it is always possible to merge DBC files via some editing tools, or just concatenating the files, but it is easier/more clean to just be able to load the dbcs you get "as is"